src/compiler/checker.ts

```diff
@@ -17946,7 +17949,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             return (type as SubstitutionType).objectFlags & ObjectFlags.IsGenericType;
         }
         return (type.flags & TypeFlags.InstantiableNonPrimitive || isGenericMappedType(type) || isGenericTupleType(type) ? ObjectFlags.IsGenericObjectType : 0) |
-            (type.flags & (TypeFlags.InstantiableNonPrimitive | TypeFlags.Index | TypeFlags.StringMapping) && !isPatternLiteralType(type) ? ObjectFlags.IsGenericIndexType : 0);
+            (type.flags & (TypeFlags.InstantiableNonPrimitive | TypeFlags.Index | TypeFlags.StringMapping) && !isPatternLiteralType(type, /*ignoreGenericIntersections*/ true) ? ObjectFlags.IsGenericIndexType : 0);

```

Comment:
alternatives that I considered:
- using other names for this parameter (`deferGenerics`, `deferGenericIntersections`?)
- writing "wrapper functions" for `isPatternLiteralType` and `isPatternLiteralPlaceholderType`, something like `isGenericPatternLiteralType` and `isGenericPatternLiteralPlaceholderType`. I'd still have to handle common concerns in those new functions though - so I concluded that an extra parameter in the existing ones makes sense the most to me.